app/source.js

```diff
@@ -574,13 +590,13 @@ window.Source = function(templates, cwd, tm, source, revlayers, examples, isMapb
             });
         }
         map.options.minZoom = this.model.get('minzoom');
-        map.options.maxZoom = 22;
+        map.options.maxZoom = 16;
         // Refresh map layer.
         if (tiles) map.removeLayer(tiles);
         tiles = L.mapbox.tileLayer({
             tiles: ['/source/{z}/{x}/{y}.png?id=' + this.model.id + '&' + mtime],
             minzoom: this.model.get('minzoom'),
-            maxzoom: 22
+            maxzoom: 16
```

Comment:
These maxzoom values controls the max zoom that the source editor allows you to preview your source at. I think it probably makes sense to leave this at z22 (ie. so you can see what your source data looks like overzoomed).

To limit the maxzoom that the sliders in the settings pane allow I think you'll want to adjust the max properties here:

https://github.com/mapbox/mapbox-studio/blob/mb-pages/templates/source.html#L91-L104